src/Stomp/Network/Connection.php

```diff
@@ -323,6 +339,7 @@ public function writeFrame(Frame $stompFrame)
         if (!@fwrite($this->connection, $data, strlen($data))) {
             throw new ConnectionException('Was not possible to write frame!', $this->activeHost);
         }
+        $this->getObserver()->transmittedFrame($stompFrame);
```

Comment:
we shouldnt use the getter inside off this class... this is a very hot path of our lib .. just use `$this->observer` instead.

I have the same feeling for `isConnected()`, `hasDataToRead()` etc. within this class, but this shouldn't be changed with this PR. I guess we can speedup the client a lot when we prevent this calls and just inline the code (the methods should still be preserved for public consumption and BC)